src/quota_aggregator_impl_test.cc

```diff
@@ -0,0 +1,354 @@
+
+  AllocateQuotaResponse error_response1_;
+  AllocateQuotaResponse error_response2_;
+
+  AllocateQuotaResponse empty_response_;
+
+  std::unique_ptr<QuotaAggregator> aggregator_;
+  std::vector<AllocateQuotaRequest> flushed_;
+};
+
+TEST_F(QuotaAggregatorImplTest, TestFirstReqeustNotFound) {
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::NOT_FOUND, aggregator_->Quota(request1_, &response));
+}
+
+TEST_F(QuotaAggregatorImplTest, TestSecondRequestFound) {
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::NOT_FOUND, aggregator_->Quota(request1_, &response));
+  EXPECT_OK(aggregator_->Quota(request1_, &response));
+  EXPECT_TRUE(MessageDifferencer::Equals(response, empty_response_));
+}
+
+TEST_F(QuotaAggregatorImplTest, TestFirstReqeustSucceeded) {
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::NOT_FOUND, aggregator_->Quota(request1_, &response));
+  EXPECT_OK(aggregator_->CacheResponse(request1_, pass_response1_));
+  EXPECT_OK(aggregator_->Quota(request1_, &response));
+  EXPECT_TRUE(MessageDifferencer::Equals(response, pass_response1_));
+}
+
+TEST_F(QuotaAggregatorImplTest, TestFirstReqeustQuotaExceed) {
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::NOT_FOUND, aggregator_->Quota(request1_, &response));
+  EXPECT_OK(aggregator_->CacheResponse(request1_, error_response1_));
+  EXPECT_OK(aggregator_->Quota(request1_, &response));
+  EXPECT_TRUE(MessageDifferencer::Equals(response, error_response1_));
+}
+
+TEST_F(QuotaAggregatorImplTest, TestNotMatchingServiceName) {
+  *(request1_.mutable_service_name()) = "some-other-service-name";
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::INVALID_ARGUMENT,
+                    aggregator_->Quota(request1_, &response));
+}
+
+TEST_F(QuotaAggregatorImplTest, TestNoOperation) {
+  request1_.clear_allocate_operation();
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::INVALID_ARGUMENT,
+                    aggregator_->Quota(request1_, &response));
+}
+
+TEST_F(QuotaAggregatorImplTest, TestFlushAggregatedRecord) {
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::NOT_FOUND, aggregator_->Quota(request1_, &response));
+  EXPECT_OK(aggregator_->CacheResponse(request1_, pass_response1_));
+  EXPECT_OK(aggregator_->Quota(request1_, &response));
+  EXPECT_TRUE(MessageDifferencer::Equals(response, pass_response1_));
+
+  EXPECT_EQ(flushed_.size(), 0);
+
+  // simulate refresh timeout
+  std::this_thread::sleep_for(std::chrono::milliseconds(110));
+
+  EXPECT_OK(aggregator_->Flush());
+  EXPECT_EQ(flushed_.size(), 1);
+}
+
+TEST_F(QuotaAggregatorImplTest, TestAggregatedRecordNoRefresh) {
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::NOT_FOUND, aggregator_->Quota(request1_, &response));
+  EXPECT_OK(aggregator_->CacheResponse(request1_, pass_response1_));
+  EXPECT_OK(aggregator_->Quota(request1_, &response));
+  EXPECT_TRUE(MessageDifferencer::Equals(response, pass_response1_));
```

Comment:
TestCacheRefreshAllAggregated is doing the similar test.
the second element is cached 50 ms after the first cached. 
Then both elements were aggregated.
After 60 ms, the first element was added to the flushed_, then another 60ms the second element was added to the flushed_